WalletWasabi.Fluent/ViewModels/Dialog/DialogViewModelBaseOfTResult.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Fluent.ViewModels.Dialog
+{
+	/// <summary>
+	/// Base ViewModel class for Dialogs that returns a value back.
+	/// </summary>
+	/// <typeparam name="TResult">The type of the value to be returned when the dialog is finished.</typeparam>
+	public abstract class DialogViewModelBase<TResult> : DialogViewModelBase
+	{
+		private TaskCompletionSource<TResult> CurrentTaskCompletionSource { get; set; }
+
+		/// <summary>
+		/// Method to be called when the dialog intends to close
+		/// and ready to pass a value back to the caller.
+		/// </summary>
+		/// <param name="value">The return value of the dialog</param>
+		public void CloseDialog(TResult value)
```

Comment:
This is never called!

Reply:
this is only called when the user wants to return a value

Reply:
It must be called everytime! null is fine! but it must be called!